src/mandelbrot.rs

```diff
@@ -50,28 +47,73 @@ impl Sub for Vecf64 {
     }
 }
 
-pub fn mbrot8(cr: Vecf64, ci: Vecf64) -> u8 {
-    let mut zr = ZEROS;
-    let mut zi = ZEROS;
-    let mut tr = ZEROS;
-    let mut ti = ZEROS;
-    for _ in 0..MAX_ITER / 5 {
-        for _ in 0..5 {
-            zi = (zr + zr) * zi + ci;
-            zr = tr - ti + cr;
-            tr = zr * zr;
-            ti = zi * zi;
+#[derive(Default)]
+pub struct Mandelbrot8 {
+    zr: Vecf64,
+    zi: Vecf64,
+    tr: Vecf64,
+    ti: Vecf64,
+
+    cr: Vecf64,
+    ci: Vecf64,
+    ci2: Vecf64,
+}
+
+impl Mandelbrot8 {
+    pub fn new(ci: Vecf64) -> Self {
+        Mandelbrot8 {
+            zr: ZEROS,
+            zi: ZEROS,
+            tr: ZEROS,
+            ti: ZEROS,
+
+            cr: ZEROS,
+            ci: ci,
+            ci2: ci * ci,
         }
-        if (tr + ti).0.iter().all(|&t| t > 4.) {
-            return 0;
+    }
+
+    pub fn run(&mut self, cr: Vecf64, cr2: Vecf64) -> u8 {
+        self.zr = cr;
+        self.zi = self.ci;
+        self.tr = cr2;
+        self.ti = self.ci2;
+        self.cr = cr;
+
+        self.advance(4);
+        for _ in 0..MAX_ITER / 5 - 1 {
```

Comment:
The fastest C version check every 7 iterations. Maybe that can improve the performances.